cli/src/commands/branch.rs

```diff
@@ -301,6 +317,44 @@ fn cmd_branch_create(
     Ok(())
 }
 
+fn cmd_branch_rename(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchRenameArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let repo = workspace_command.repo().as_ref();
+    let view = repo.view();
+    let old_branch = &args.old;
+    // call clone because .set_local_branch_target takes ownership
+    // of the target_commit, so we need to create a copy that is no
+    // longer owned by the workspace.
+    let target_commit = view.get_local_branch(old_branch).clone();
```

Comment:
nit: I'd call it just `target` (or maybe `ref_target`) since it can point to multiple commits (or no commit).

Reply:
I went with `ref_target`. This (at least for me), drives home the point that it doesn't map 1:1 to a single commit.